eth/watchers/roundswatcher.go

```diff
@@ -0,0 +1,160 @@
+package watchers
+
+import (
+	"fmt"
+	"math/big"
+	"sync"
+
+	ethcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/event"
+	"github.com/livepeer/go-livepeer/eth/contracts"
+
+	"github.com/golang/glog"
+	"github.com/livepeer/go-livepeer/eth"
+	"github.com/livepeer/go-livepeer/eth/blockwatch"
+)
+
+// RoundsWatcher is a type alias for a thread safe in-memory cache that watches on-chain events for new initialized rounds
+type RoundsWatcher struct {
+	mu                   sync.RWMutex
+	lastInitializedRound *big.Int
+	blockHashForRound    [32]byte
+
+	quit chan struct{}
+
+	watcher Watcher
+	lpEth   eth.LivepeerEthClient
+	dec     *EventDecoder
+}
+
+type Watcher interface {
```

Comment:
Can we move this to a separate file since it is used by multiple event watcher implementations and call it `BlockWatcher`? Maybe a `types.go` file.

Reply:
fixed in 80d3e25